examples/tutorial4/tutorial/app.py

```diff
@@ -97,20 +86,20 @@ def draw_text(self):
         y = 185
         font = toga.Font(family='sans-serif', size=20)
         width, height = font.measure('Tiberius', tight=True)
-        with self.canvas.stroke():
+        with self.canvas.stroke(line_width=4.0):
             self.canvas.rect(x - 10, y - height + 2, width, height + 2)
-        with self.canvas.fill():
-            self.canvas.fill_style('rgba(149.0, 119, 73, 1)')
+        with self.canvas.fill(color='rgba(149.0, 119, 73, 1)'):
             self.canvas.write_text('Tiberius', x, y, font)
 
-    def draw_tiberius(self, canvas, context):
-        self.canvas.set_context(context)
-        self.fill_head()
-        self.draw_eyes()
-        self.draw_horns()
-        self.draw_nostrils()
-        self.stroke_head()
-        self.draw_text()
+    def draw_tiberius(self):
+        tiberius_context = self.canvas.create_context()
+        with self.canvas.context(tiberius_context):
```

Comment:
Why is the context abstraction needed here? Is this just a function of the example, or is it actually required?

Reply:
No, it's not required. In the tutorial I said it was optional. I think there is work to do to make the API documentation even more clear on how to use Canvas for the different use cases, so I'll work on that.